### PR TITLE
Add ORC bloom filter support for DATE, TIMESTAMP and REAL

### PR DESCRIPTION
**Pull request by dain**

### CODE REVIEW

File: presto-orc/src/main/java/io/prestosql/orc/metadata/statistics/BloomFilter.java [hunk 1]
```diff
@@ -361,6 +369,56 @@ public static long hash64(byte[] data)
             return hash;
         }
 
+        @SuppressWarnings("fallthrough")
+        public static long hash64(Slice data)
```
Comment (martint) [inline presto-orc/src/main/java/io/prestosql/orc/metadata/statistics/BloomFilter.java]: Who calls this method? The method `testSlice` above calls `OrcMurmur3.hash64`

Comment (martint) [reply presto-orc/src/main/java/io/prestosql/orc/metadata/statistics/BloomFilter.java]: Got it. I missed that. It wasn't obvious from the diff github shows.

File: presto-orc/src/main/java/io/prestosql/orc/StripeReader.java [hunk 2]
```diff
@@ -232,7 +232,8 @@ private static boolean isSupportedStreamType(Stream stream, OrcTypeKind orcTypeK
     {
         if (stream.getStreamKind() == BLOOM_FILTER) {
             // non-utf8 bloom filters are not allowed for character types
-            return orcTypeKind != OrcTypeKind.STRING && orcTypeKind != OrcTypeKind.VARCHAR && orcTypeKind != OrcTypeKind.CHAR;
+            // non-utf8 bloom filters are not supported for timestamp
```
Comment (electrum) [inline presto-orc/src/main/java/io/prestosql/orc/StripeReader.java]: Should this be "character types or timestamp"

Comment (dain) [reply presto-orc/src/main/java/io/prestosql/orc/StripeReader.java]: The line above says "non-utf8 bloom filters are not allowed for character types"

File: presto-orc/src/test/java/io/prestosql/orc/TestOrcBloomFilters.java [hunk 3]
```diff
@@ -251,6 +262,7 @@ public void testExtractValuesFromSingleDomain()
                 .put(DATE, 901L)
                 .put(BIGINT, 4321L)
                 .put(DOUBLE, 0.123)
+                .put(REAL, (long) (floatToIntBits(0.456f)))
```
Comment (findepi) [inline presto-orc/src/test/java/io/prestosql/orc/TestOrcBloomFilters.java]: `floatToRawIntBits`, unnecessary braces

File: presto-orc/src/test/java/io/prestosql/orc/TestReadBloomFilter.java [hunk 4]
```diff
@@ -64,6 +66,7 @@ public void test()
 
         testType(DATE, ImmutableList.of(new SqlDate(1), new SqlDate(5_000), new SqlDate(10_000)), 5_000L, 7_777L);
 
+        testType(REAL, ImmutableList.of(1.11f, 500_000.56f, 1_000_000.99f), (long) floatToIntBits(500_000.56f), (long) floatToIntBits(777_777.77f));
```
Comment (findepi) [inline presto-orc/src/test/java/io/prestosql/orc/TestReadBloomFilter.java]: `floatToRawIntBits`

**Pull request merged**